src/ScillaParser.ts

```diff
@@ -173,3 +178,31 @@ const extractTransitions = (ccompsElem: any[]): Transitions => {
     };
   });
 };
+
+function parseField(row : any):Field{
+  const field_type = row[0];
+
+  if (field_type == "PrimType"){
+    const type = row[1];
+    return {
+      name: "",
+      typeobject: type,
+      type: type
+    }
+  } else if (field_type == "ADT"){
+    const ctor = row[1][1][1];
+    const argtypes = row[2].map(parseField);
+    const name = row[0][1][1];
+    const ADT : ADTField = {
+      ctor: ctor,
+      argtypes: argtypes
+    }
+    return {
+      name: name,
+      typeobject: ADT,
+      type: ctor + argtypes.map((arg:Field)=>` (${arg.type})`).join('')
+    }
+  } else {
+    throw new Error(`Encountered unexpected field type ${row}`);
```

Comment:
Can you throw [HardhatPluginError](https://hardhat.org/hardhat-runner/docs/advanced/building-plugins#throwing-errors-from-your-plugins)?

Reply:
I will make the change.

Reply:
@its-saeed Should be fixed now.